fem/transfer.hpp

```diff
@@ -158,11 +158,11 @@ class InterpolationGridTransfer : public GridTransfer
     defined in physical space and, generally, vary between different mesh
     elements.
 
-    This class currently only fully supports L2 finite element spaces and fine
-    meshes that are a uniform refinement of the coarse mesh. Generally, the
-    coarse and fine FE spaces can have different orders, however, in order for
-    the backward operator to be well-defined, the number of the fine dofs (in a
-    coarse element) should not be smaller than the number of coarse dofs.
+    This class supports L2 finite element spaces and fine meshes that are a
+    uniform refinement of the coarse mesh. Generally, the coarse and fine FE
+    spaces can have different orders, however, in order for the backward
+    operator to be well-defined, the number of the fine dofs (in a coarse
+    element) should not be smaller than the number of coarse dofs.
 
     If used on H1 finite element spaces, the transfer will be performed locally,
```

Comment:
Maybe I am not down far enough in this file, but there is still functionality that let's you use the L2 FE space L2_projection, but using H1 fields. Is there an instance where you want to do this? Should we add a message so that a user realizes they "may" be using the L2 discontinuous mapping option for their H1 field? Or if you try this it actually uses the H1 continuous mapping implementation that we are reviewing? @ebchin @tzanio please clarify (I'm curious).

Reply:
I can see how this might cause confusion or lead a user to do something unintended.  I clarified the comments for the `L2ProjectionGridTransfer` and `L2ProjectionH1GridTransfer` classes so it's more clear which one is for L2 spaces and which one is for H1 spaces.  I'm not sure we need to explicitly disallow `L2ProjectionGridTransfer` on H1 spaces.  Maybe a user wants to avoid a global solve and doesn't need mass conservation?  Though the collocation approach in `InterpolationGridTransfer` might be more appropriate in that case...  Any thoughts @tzanio or @pazner ?

Reply:
Fixed in latest commit.